src/backend/columnar/columnar_metadata_tables.c

```diff
@@ -977,23 +980,42 @@ DeleteMetadataRows(RelFileNode relfilenode)
 		return;
 	}
 
-	ScanKeyInit(&scanKey[0], Anum_columnar_stripe_storageid,
-				BTEqualStrategyNumber, F_INT8EQ, UInt64GetDatum(metapage->storageId));
+	DeleteStorageFromColumnarMetadataTable(ColumnarStripeRelationId(),
+										   metapage->storageId);
+	DeleteStorageFromColumnarMetadataTable(ColumnarChunkGroupRelationId(),
+										   metapage->storageId);
+	DeleteStorageFromColumnarMetadataTable(ColumnarChunkRelationId(),
+										   metapage->storageId);
+}
 
-	Oid columnarStripesOid = ColumnarStripeRelationId();
-	Relation columnarStripes = try_relation_open(columnarStripesOid, AccessShareLock);
-	if (columnarStripes == NULL)
+
+/*
+ * DeleteStorageFromColumnarMetadataTable removes the rows with given
+ * storageId from given columnar metadata table.
+ */
+static void
+DeleteStorageFromColumnarMetadataTable(Oid metadataTableId, uint64 storageId)
+{
+	ScanKeyData scanKey[1];
+	AttrNumber storageIdAtrrNumber =
```

Comment:
I suggest you pass `storageIdAttrNumber` and `indexOid` as argument to this function. Think about the situation you have a new metadata table. Now, you have to be careful to update three functions and it is easy to miss one of them. But if you pass these ids as arguments, you only modify one place and it is more difficult to miss.